model/permission/permissions.go

```diff
@@ -455,12 +470,13 @@ func CreateShareSet(db prefixer.Prefixer, parent *Permission, sourceID string, c
 }
 
 // CreateSharePreviewSet creates a Permission doc for previewing a sharing
-func CreateSharePreviewSet(db prefixer.Prefixer, sharingID string, codes map[string]string, set Set) (*Permission, error) {
+func CreateSharePreviewSet(db prefixer.Prefixer, sharingID string, codes map[string]string, set Set, metadata *metadata.CozyMetaData) (*Permission, error) {
```

Comment:
Why didn't you choose the same solution for giving metadata for both `CreateShareSet` and `CreateSharePreviewSet`? In one case, there is a single parameter `subdoc` (a struct with a field for the set and another field for the metadata), and in the other case, there are 2 parameters.